config/application.rb

```diff
@@ -26,8 +26,8 @@ class Application < Rails::Application
 
     config.x.registry_url        = ENV["DOCKER_REGISTRY_URL"] || "http://localhost:5000"
     config.x.no_ssl_verification = ENV["NO_SSL_VERIFICATION"].in? %w(1 true yes)
-    config.x.basic_auth_user     = ENV["BASIC_AUTH_USER"]
-    config.x.basic_auth_password = ENV["BASIC_AUTH_PASSWORD"]
+    config.x.basic_auth_user     = File.file?("/run/secrets/BASIC_AUTH_USER") ? File.read("/run/secrets/BASIC_AUTH_USER").strip : ENV["BASIC_AUTH_USER"]
+    config.x.basic_auth_password = File.file?("/run/secrets/BASIC_AUTH_PASSWORD") ? File.read("/run/secrets/BASIC_AUTH_PASSWORD").strip : ENV["BASIC_AUTH_PASSWORD"]
```

Comment:
Could you please extract this logic in a new module?
Something like this:

```ruby
# lib/config.rb
module Config
  extend self

  def get(name)
    if File.file?("/run/secrets/#{name}")
      File.read("/run/secrets/#{name}").strip
    else
      ENV[name]
    end
  end
end

# config/application.rb
config.x.basic_auth_user     = Config.get("BASIC_AUTH_USER")
config.x.basic_auth_password = Config.get("BASIC_AUTH_PASSWORD")
```

I'd then also allow using this secrets feature for all configs and not just the htauth-credentails to avoid confusion for users.